### ISSUE TITLE
Run `pdm add ...` with toml comments will get a error

### ISSUE THREAD

**Issue by abersheeran** [pdm-project/pdm#462]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

`pyproject.toml`: 

```
[project]
dependencies = [
  "index.py~=0.18",
  "uvicorn~=0.13",
  "uvloop~=0.15", # 提速 uvicorn
  "httptools~=0.2", # 提速 uvicorn
  "elasticsearch[async]~=7.12",
  "aioredis~=1.3",
  "aiokafka~=0.7",
]
requires-python = ">=3.9,<4.0"

[tool.pdm.scripts]
_.env_file = ".env"
```

Then run `pdm add protobuf`

```
Adding packages to default dependencies: protobuf
✔ 🔒 Lock successful
Changes are written to pdm.lock.
Changes are written to pyproject.toml.
[UnexpectedCharError]: Unexpected character: ',' at line 7 col 0
Add '-v' to see the detailed traceback
```

`pyproject.toml`: 

```
[project]
dependencies = [
    "index.py~=0.18",
    "uvicorn~=0.13",
    "uvloop~=0.15",
     # 提速 uvicorn
,
    "httptools~=0.2",
     # 提速 uvicorn
,
    "elasticsearch[async]~=7.12",
    "aioredis~=1.3",
    "aiokafka~=0.7",
    "protobuf~=3.17",
]
requires-python = ">=3.9,<4.0"

[tool.pdm.scripts]
_.env_file = ".env"
```

## Environment Information

```bash
PDM version:        1.5.3                                                                
Python Interpreter: /usr/bin/python3.9 (3.9)                                             
Project Root:       *******************                   
Project Packages:   **********************
{
  "implementation_name": "cpython",
  "implementation_version": "3.9.0",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "4.4.0-19041-Microsoft",
  "platform_system": "Linux",
  "platform_version": "#488-Microsoft Mon Sep 01 13:43:00 PST 2020",
  "python_full_version": "3.9.0+",
  "platform_python_implementation": "CPython",
  "python_version": "3.9",
  "sys_platform": "linux"
}
```

**Comment by frostming** [comment]
add -v to see the verbose traceback? I doubt it is a tomlkit issue.

**Comment by abersheeran** [comment]
```
Traceback (most recent call last):
  File "/home/aber/.local/bin/pdm", line 8, in <module>
    sys.exit(main())
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 184, in main
    return Core().main(args)
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 145, in main
    raise cast(Exception, err).with_traceback(traceback)
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/core.py", line 140, in main
    f(options.project, options)
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/cli/commands/add.py", line 35, in handle
    actions.do_add(
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/cli/actions.py", line 179, in do_add
    project.write_lockfile(lockfile, False)
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/project/core.py", line 369, in write_lockfile
    toml_data["metadata"].update(self.get_lock_metadata())
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/project/core.py", line 364, in get_lock_metadata
    content_hash = tomlkit.string("sha256:" + self.get_content_hash("sha256"))
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/project/core.py", line 417, in get_content_hash
    "sources": self.tool_settings.get("source", []),
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/project/core.py", line 105, in tool_settings
    data = self.pyproject
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/pdm/project/core.py", line 95, in pyproject
    data = tomlkit.parse(self.pyproject_file.read_text("utf-8"))
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/tomlkit/api.py", line 51, in parse
    return Parser(string).parse()
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/tomlkit/parser.py", line 160, in parse
    key, value = self._parse_table()
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/tomlkit/parser.py", line 1152, in _parse_table
    item = self._parse_item()
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/tomlkit/parser.py", line 306, in _parse_item
    return self._parse_key_value(True)
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/tomlkit/parser.py", line 397, in _parse_key_value
    val = self._parse_value()
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/tomlkit/parser.py", line 569, in _parse_value
    return self._parse_array()
  File "/home/aber/.local/pipx/venvs/pdm/lib/python3.8/site-packages/tomlkit/parser.py", line 737, in _parse_array
    raise self.parse_error(UnexpectedCharError, self._current)
tomlkit.exceptions.UnexpectedCharError: Unexpected character: ',' at line 7 col 0
```

**Comment by frostming** [comment]
Note that even with the fix in 65dcd1e, TOML document will still be broken because of the existence of the trailing comma before the closing bracket of the array, it is a bug with tomlkit.

For the workaround, you have to remove the trailing comma after the last item.
